tomplot/local_area_plot.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+
+
+def area_restriction(field_data, coords_X, coords_Y, Y_lim=None, X_lim=None):
+    """
+    a function that takes a data dictionary and will restrict it for a given value
+    For the coordinate fields you must pass in a 1d object so either coords[:,level]
+    for slicing on z or coords.flatten if slicing along lat / lon
+
+    args:
+    field_data: The Plotting field you wish to return
+    coords_X: The coordinates you wish to plot along the X axis.
+    coords_Y: The field you wish to plot along the Y axis.
+    X_lim: tuple containing the (lower, upper) limits for the X axis
+    Y_lim: tuple containing the (lower, upper) limits for the Y axis
+    returns
+    restricted field data, X_coords, Y_coords
+    """
+    data_dict = {'field': field_data, 'X': coords_X, 'Y': coords_Y}
+    df = pd.DataFrame(data_dict)
+    # ive used this logic to allow a user to only enter a limit for one axis
```

Comment:
I think your logic is fine! It's just if we want to combine the two arguments. Remember to remove this comment either way.